coderd/dormancy/dormantusersjob.go

```diff
@@ -0,0 +1,61 @@
+package dormancy
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+
+	"github.com/coder/coder/coderd/database"
+)
+
+const (
+	jobInterval           = 5 * time.Minute
+	accountDormancyPeriod = 90 * 24 * time.Hour
+)
+
+func CheckInactiveUsers(ctx context.Context, logger slog.Logger, db database.Store) func() {
+	return CheckInactiveUsersWithOptions(ctx, logger, db, jobInterval, accountDormancyPeriod)
+}
+
+func CheckInactiveUsersWithOptions(ctx context.Context, logger slog.Logger, db database.Store, checkInterval, dormancyPeriod time.Duration) func() {
+	logger = logger.Named("dormancy")
+
+	ctx, cancelFunc := context.WithCancel(ctx)
+	done := make(chan struct{})
+	ticker := time.NewTicker(checkInterval)
+	go func() {
+		defer close(done)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+			}
+
+			lastSeenAfter := database.Now().Add(-dormancyPeriod)
+			logger.Debug(ctx, "check inactive user accounts", slog.F("dormancy_period", dormancyPeriod), slog.F("last_seen_after", lastSeenAfter))
+
+			updatedUsers, err := db.UpdateInactiveUsersToDormant(ctx, lastSeenAfter)
+			if err != nil && !xerrors.Is(err, sql.ErrNoRows) {
+				logger.Error(ctx, "can't mark inactive users as dormant", slog.Error(err))
+				goto done
```

Comment:
suggestion: `continue` instead; logging that we checked 0 users isn't really useful here.

Reply:
Fixed